src/server/mod.rs

```diff
@@ -35,22 +34,23 @@ pub struct SpaceapiServer {
 
 impl SpaceapiServer {
 
-    pub fn new<T>(host: Ipv4Addr, port: u16,
-                  status: api::Status,
-                  redis_connection_info: T,
-                  status_modifiers: Vec<Box<modifiers::StatusModifier>>)
-                  -> Result<SpaceapiServer, errors::SpaceapiServerError>
-                  where T: IntoConnectionInfo {
-
-        Ok(SpaceapiServer {
-            host: host,
-            port: port,
-            status: status,
-            redis_connection_info: try!(redis_connection_info.into_connection_info()),
-            sensor_specs: Arc::new(Mutex::new(vec![])),
-            status_modifiers: status_modifiers,
-        })
-    }
+    pub fn new<T, U>(socket_addr: U,
+                     status: api::Status,
+                     redis_connection_info: T,
+                     status_modifiers: Vec<Box<modifiers::StatusModifier>>)
```

Comment:
Would make sense to reorder them since they appear reverse in the argument list.